cookieJar.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+/**
+ * Class for maintaining a cookie jar between http requests
+ * @class CookieJar
+ */
+var CookieJar = function() {
+  this.cookies = {};
+};
+
+/**
+ * Serializes stored cookies into http header format
+ * @return {String}
+ */
+CookieJar.prototype.get = function() {
+  return Object.keys(this.cookies)
+    .map(function(key) {
+      var str = '';
+      str += key;
+      if(this.cookies[key]) {
+        str += '=' + this.cookies[key];
+      }
+      return str;
+    }, this)
+    .join('; ');
+};
+
+/**
+ * Parses a single header line of cookie, parses them, and stores them in a hash
+ * @param  {String} h Header line
+ */
+CookieJar.prototype.parseCookieHeader = function(h) {
```

Comment:
nit: there's no need to use a one letter variable name. `headerLine` would be better